src/components/TreeSelect/TreeSelect.tsx

```diff
@@ -82,6 +86,7 @@ export const TreeSelect = React.forwardRef(function TreeSelect<T>(
         selectedById: selected,
     });
 
+    // used for control render. Needed a smart way to share list context here
     const listParsedState = useList({
```

Comment:
Here is one big problem: we have to call  `useList` to get same result as inside `TreeList` component.
I will be glad to have ideas on how this can be improved. Perhaps not in this task